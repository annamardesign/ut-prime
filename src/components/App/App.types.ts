import {ThemeOptions} from '@material-ui/core/styles';
import React from 'react';
import {Store} from 'redux';

import {menuItem, showTab} from '../Context';

export interface utTheme extends ThemeOptions {
    ut: {
        classes: {}
    }
}

export interface Props extends React.HTMLAttributes<HTMLDivElement> {
    menu: menuItem[],
    showTab: showTab,
    theme: utTheme,
    portalName: string,
    store: Store
}

export type StyledType = React.FC<Props>
