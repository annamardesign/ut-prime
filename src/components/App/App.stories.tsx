import React from 'react';
import { withReadme } from 'storybook-readme';
import { action } from '@storybook/addon-actions';

// @ts-ignore: md file and not a module
import README from './README.md';
import App from './index';

export default {
    title: 'App',
    component: App,
    decorators: [withReadme(README)]
};

export const Basic: React.FC<{}> = () => <App onClick={action('clicked')}>Hello App</App>;
