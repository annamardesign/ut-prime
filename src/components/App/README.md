# Button

## How to use

```jsx
import App from 'ut-front-devextreme/core/App';

<App onClick={handlerOnClick}>Hello App</App>;
```

## Props

- **className** - class applied to root `App` html element.

| propName  | propType | defaultValue | isRequired |
| --------- | -------- | ------------ | ---------- |
| className | string   |              | no         |
