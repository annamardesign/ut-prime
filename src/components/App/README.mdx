# App
import { Canvas, Story, ArgsTable } from "@storybook/addon-docs";

The application component, usually passed to ReactDOM.render

## How to use

```jsx
import App from 'ut-prime/core/App';

<App
    portalName='test app'
    state={state}
    theme={{
        ut: {
            classes: {}
        },
        palette: {
            type: 'dark'
        }
    }}
/>
```

<Canvas>
    <Story id="app--basic" />
</Canvas>

## Properties

<ArgsTable story="."/>
