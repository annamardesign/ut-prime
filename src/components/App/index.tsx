import React from 'react';
import {Provider} from 'react-redux';
import {Router, Route, Switch} from 'react-router';
import {hot} from 'react-hot-loader';
import {createHashHistory, createMemoryHistory} from 'history';
import CssBaseline from '@material-ui/core/CssBaseline';
import { ThemeProvider, createMuiTheme } from '@material-ui/core/styles';
import 'devextreme/dist/css/dx.common.css';
import 'devextreme/dist/css/dx.light.css';

import LoginPage from '../Login';
import Main from '../Main';
import Context from '../Context';

import { StyledType } from './App.types';
import PageNotFound from './PageNotFound';

const App: StyledType = ({menu, store, theme, portalName, showTab}) =>
    <Provider store={store}>
        <ThemeProvider theme={createMuiTheme(theme)}>
            <CssBaseline />
            <Context.Provider value={{menu, portalName, showTab}}>
                <Router history={(typeof window !== 'undefined') ? createHashHistory() : createMemoryHistory()}>
                    <Switch>
                        <Route path='/login' component={LoginPage} />
                        <Route path='/sso/:appId/:ssoOrigin/login' component={LoginPage} />
                        <Route component={Main} />
                        <Route path='*' component={PageNotFound} />
                    </Switch>
                </Router>
            </Context.Provider>
        </ThemeProvider>
    </Provider>;

export default hot(module)(App);
