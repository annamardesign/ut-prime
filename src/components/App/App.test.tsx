import React from 'react';
import {createStore} from 'redux';

import { render } from '../test';
import App from './index';

describe('<App />', () => {
    it('render equals snapshot', () => {
        const store = createStore(() => ({
            login: {
                get() {}
            }
        }));
        const { getByTestId } = render(
            <App
                menu={[]}
                portalName='test'
                showTab={() => {}}
                store={store}
                theme={{
                    ut: {
                        classes: {}
                    }
                }}
            />
        );
        expect(getByTestId('ut-front-test')).toMatchSnapshot();
    });
});
