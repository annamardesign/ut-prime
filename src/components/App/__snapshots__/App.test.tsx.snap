// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render equals snapshot 1`] = `
<div
  data-testid="ut-front-test"
/>
`;
